WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewModelProtocol.swift

```diff
@@ -0,0 +1,81 @@
+import Yosemite
+
+/// A view model for `ProductFormViewController` to add/edit a generic product model (e.g. `Product` or `ProductVariation`).
+///
+protocol ProductFormViewModelProtocol {
+    associatedtype ProductModel: ProductFormDataModel & TaxClassRequestable
+
+    /// Emits product on change, except when the product name is the only change (`productName` is emitted for this case).
+    var observableProduct: Observable<ProductModel> { get }
+
+    /// Emits product name on change. If product name is not editable, `nil` is returned.
```

Comment:
🤔

Reply:
sorry it sounds confusing not knowing the differences between editing a `Product` and `ProductVariation`! since `ProductVariation`'s name is derived from its attributes (the option values in the variation), it's not editable like `Product`. I updated the comment here to hopefully clarify a bit: https://github.com/woocommerce/woocommerce-ios/pull/2565/commits/cfb7811f75f987a73e60e94c8aa0636e79c5d5df